tests/ui/missing_const_for_fn/could_be_const.rs

```diff
@@ -0,0 +1,55 @@
+#![warn(clippy::missing_const_for_fn)]
+#![allow(clippy::let_and_return)]
+
+use std::mem::transmute;
+
+struct Game {
+    guess: i32,
+}
+
+impl Game {
+    // Could be const
+    pub fn new() -> Self {
+        Self { guess: 42 }
+    }
+}
+
+// Could be const
+fn one() -> i32 {
+    1
+}
+
+// Could also be const
+fn two() -> i32 {
+    let abc = 2;
+    abc
+}
+
+// TODO: Why can this be const? because it's a zero sized type?
+// There is the `const_string_new` feature, but it seems that this already works in const fns?
+fn string() -> String {
+    String::new()
+}
+
+// Could be const
+unsafe fn four() -> i32 {
+    4
+}
+
+// Could also be const
+fn generic<T>(t: T) -> T {
+    t
+}
+
+// FIXME: This could be const but is currently not linted
+fn sub(x: u32) -> usize {
+    unsafe { transmute(&x) }
```

Comment:
transmute needs the `const_transmute` feature gate

Reply:
It looks like it also depends on the `const_fn` feature gate? I wasn't able to make the lint suggest to make this a const fn. I tried enabling both `const_transmute` and `const_fn` and it didn't detect this as a const fn. I guess `is_min_const_fn` doesn't know anything about the `const_transmute` feature?

Reply:
Yea, and that makes sense imo. We only want to suggest to make things `const fn` if that would work on stable